apps/studio/src/features/editing-experience/components/ScheduledPublishingModal/ScheduledPublishingModal.tsx

```diff
@@ -0,0 +1,427 @@
+                        title="Publish now"
+                        description="Changes will be live on your site in approximately 5-10 minutes."
+                        isSelected={publishMode === PublishMode.NOW}
+                        onSelect={() =>
+                          methods.setValue("publishMode", PublishMode.NOW)
+                        }
+                      />
+                      <PublishModeCard
+                        icon={BiTimeFive}
+                        title="Publish later"
+                        description="Let us know when the page should start publishing."
+                        isSelected={publishMode === PublishMode.SCHEDULED}
+                        onSelect={() =>
+                          methods.setValue("publishMode", PublishMode.SCHEDULED)
+                        }
+                      />
+                    </VStack>
+                  </VStack>
+                  {publishMode === PublishMode.SCHEDULED && (
+                    <SchedulePublishDetails />
+                  )}
+                </VStack>
+                {publishMode === PublishMode.SCHEDULED &&
+                  isScheduledPublishValid && <SchedulePublishBanner />}
+              </VStack>
+            </FormProvider>
+          </ModalBody>
+          <ModalFooter>
+            <Button
+              mr={3}
+              onClick={onClose}
+              variant="clear"
+              color="base.content.strong"
+            >
+              No, don't publish
+            </Button>
+            <Button type="submit" isLoading={isPublishingNow || isScheduling}>
+              {publishMode === PublishMode.NOW
+                ? "Publish now"
+                : "Schedule publish"}
+            </Button>
+          </ModalFooter>
+        </ModalContent>
+      </form>
+    </Modal>
+  )
+}
+
+/**
+ * if the date provided is equal to the earliestSchedule's date, the earliest allowable time should be set to the FIRST
+ * time slot after the current minimum allowable time
+ * @param selectedDate Date selected inside the datepicker
+ * @param earliestSchedule Earliest schedule time, based on the current date and MINIMUM_SCHEDULE_LEAD_TIME_MINUTES
+ * @returns
+ */
+const getEarliestAllowableTime = (
+  selectedDate: Date,
+  earliestSchedule: Date,
+): { hours: number; minutes: number } | null => {
+  if (isSameDay(selectedDate, earliestSchedule)) {
+    return {
+      hours: earliestSchedule.getHours(),
+      minutes: earliestSchedule.getMinutes(),
+    }
+  }
+  return null
+}
+
+const SchedulePublishDetails = () => {
+  const {
+    resetField,
+    watch,
+    control,
+    formState: { errors },
+  } = useFormContext<z.input<typeof schedulePublishClientSchema>>()
+
+  const publishDate = watch("publishDate")
+  const publishTime = watch("publishTime")
+  // this is the earliest date and time that the user can schedule a publish for
+  const { earliestAllowableTime, earliestSchedule } = useMemo(() => {
```

Comment:
question: what happens when user open this modal and then AFK and come back after 10 mins? are we hoping to not refresh the UI here and solely rely on backend to validate? _(to be clear, im okay with it since its trivial, but just want to confirm that its intended)_